rskj-core/src/main/java/co/rsk/net/messages/MessageType.java

```diff
@@ -260,6 +260,37 @@ public Message createMessage(BlockFactory blockFactory, RLPList list) {
 
             return new BlockReceiptsResponseMessage(id, receipts);
         }
+    },
+
+    TRANSACTION_INDEX_REQUEST_MESSAGE(103) {
+        @Override
+        public Message createMessage(BlockFactory blockFactory, RLPList list) {
+            RLPList message = (RLPList)RLP.decode2(list.get(1).getRLPData()).get(0);
+
+            byte[] rlpId = list.get(0).getRLPData();
+            long id = rlpId == null ? 0 : BigIntegers.fromUnsignedByteArray(rlpId).longValue();
+            byte[] txHash = message.get(0).getRLPData();
+            return new TransactionIndexRequestMessage(id, txHash);
+        }
+    },
+    TRANSACTION_INDEX_RESPONSE_MESSAGE(104) {
+        @Override
+        public Message createMessage(BlockFactory blockFactory, RLPList list) {
+            RLPList message = (RLPList)RLP.decode2(list.get(1).getRLPData()).get(0);
+
+            byte[] rlpId = list.get(0).getRLPData();
+            long id = rlpId == null ? 0 : BigIntegers.fromUnsignedByteArray(rlpId).longValue();
+
+            byte[] blockNumberRLP = message.get(0).getRLPData();
+            long blockNumber = blockNumberRLP == null ? 0 : BigIntegers.fromUnsignedByteArray(blockNumberRLP).longValue();
+
+            byte[] blockHash = message.get(1).getRLPData();
```

Comment:
Finally we are going to let without null check. The NodeMessageHandler class catches these exceptions.